addons/hw_drivers/websocket_client.py

```diff
@@ -40,17 +42,18 @@ def send_to_controller(device_type, params):
 
 def on_message(ws, messages):
     """
-        When a message is receive, this function is triggered
-        The message is load and if its type is 'iot_action', is sent to the device
+        When a message is received, this function is triggered.
+        The message type is checked against the WebsocketMessageHandlers,
+        and if it exists then the handler is executed.
     """
     messages = json.loads(messages)
     for message in messages:
-        if message['message']['type'] == 'iot_action':
-            payload = message['message']['payload']
-            if helpers.get_mac_address() in payload['iotDevice']['iotIdentifiers']:
-                for device in payload['iotDevice']['identifiers']:
-                    if device['identifier'] in main.iot_devices:
-                        main.iot_devices[device["identifier"]].action(payload)
+        message_type = message['message']['type']
+        handler = getattr(websocket_handlers, message_type, None)
```

Comment:
I personally find it really elegant this way :)
But, I do wonder if there is not any security risks implied as getattr retrieve any attribute of the module. Would it be possible to craft a message_type using the imported libraries from wsèhandler file, like `helpers.restart_odoo_or_reboot` would you mind testing out please ? As we let the customer the full choice of the message, we must be very careful.
If they can craft an arbitrary request, then it would be safer to check if the message_type make sense with something like:
https://stackoverflow.com/a/139198/16356970